akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -134,6 +133,16 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
 
       private val controlFrameBuffer: mutable.Queue[FrameEvent] = new mutable.Queue[FrameEvent]
       private val sendableOutstreams: mutable.Queue[Int] = new mutable.Queue[Int]
+      def enqueueOutstream(streamId: Int): Unit = {
+        if (isDebugEnabled) require(!sendableOutstreams.contains(streamId), s"Stream [$streamId] was enqueued multiple times.") // requires expensive scanning -> avoid in production
+        sendableOutstreams.enqueue(streamId)
+      }
+      def removeOutStream(streamId: Int): Unit = {
+        // in 2.12.x there's no Queue.-=, when 2.12.x support is dropped, this can be
+        // changed to Queue.-=
+        val idx = sendableOutstreams.indexOf(streamId)
+        if (idx >= 0) sendableOutstreams.remove(idx)
```

Comment:
There's one invocation of `closeStream` left in `IncomingStreamBuffer` that may be called regardless of whether a stream was previously enqueued or not, so we need to take this into consideration here. We don't want to have it happen too often, though, since scanning the whole list just to find nothing is really expensive.